skimage/restoration/j_invariant.py

```diff
@@ -143,10 +143,14 @@ def denoise_invariant(image, denoise_function, *, stride=4,
     Examples
     --------
     >>> import skimage
-    >>> from skimage.restoration import denoise_invariant, denoise_tv_chambolle
+    >>> from skimage.restoration import denoise_invariant, denoise_wavelet
     >>> image = skimage.util.img_as_float(skimage.data.chelsea())
     >>> noisy = skimage.util.random_noise(image, var=0.2 ** 2)
-    >>> denoised = denoise_invariant(noisy, denoise_function=denoise_tv_chambolle)
+
+    In this case, `denoise_wavelet` requires the optional dependency PyWavelets.
+
+    >>> import pytest; _ = pytest.importorskip("pywt")
```

Comment:
Ideally, we could hide this line altogether. 

@pllim does [doctest-plus](https://github.com/scientific-python/pytest-doctestplus) support something like that? I don't see an directive at a first glance.

Reply:
Maybe you want `.. doctest-requires::` documented in https://github.com/scientific-python/pytest-doctestplus#doctest-dependencies ?

Reply:
That seems like a good option and in my book its an argument in favor of using doctest-plus! Thanks for the quick reply. :)